utils/s2n_random.c

```diff
@@ -466,6 +466,9 @@ S2N_RESULT s2n_rand_cleanup_thread(void)
 
     s2n_per_thread_rand_state.drbgs_initialized = false;
 
+    /* Unset the thread-local destructor */
+    pthread_setspecific(s2n_per_thread_rand_state_key, NULL);
```

Comment:
You don't want to delete the key here since this is per-thread, but you definitely want to delete the key on library unload.  See https://github.com/aws/s2n-tls/issues/3990